File: loic.go
package main

import (
	"archive/zip"
	"bufio"
	"bytes"
	"fmt"
	"io"
	"net/http"
	"os"
	"os/exec"
	"path/filepath"
	"runtime"
	"strings"
)

const (
	gitRepo   = "https://github.com/NewEraCracker/LOIC.git"
	gitBranch = "master"
)

var (
	homeDir    string
	loicDir    string
	osArch     string
)

func init() {
	// Get home directory
	home, err := os.UserHomeDir()
	if err != nil {
		fmt.Println("Error getting home directory:", err)
		os.Exit(1)
	}
	homeDir = home
	loicDir = filepath.Join(homeDir, "LOIC")
	
	// Get OS/Architecture
	osArch = runtime.GOOS + "/" + runtime.GOARCH
}

func main() {
	if len(os.Args) != 2 {
		printUsage()
		os.Exit(1)
	}

	command := strings.ToLower(os.Args[1])

	switch command {
	case "install":
		installLOIC()
	case "update":
		updateLOIC()
	case "run":
		runLOIC()
	default:
		fmt.Printf("Unknown command: %s\n", command)
		printUsage()
		os.Exit(1)
	}
}

func printUsage() {
	fmt.Println("LOIC Installer for Linux/Termux")
	fmt.Println(strings.Repeat("=", 40))
	fmt.Println()
	fmt.Println("Usage: go run loic.go <install|update|run>")
	fmt.Println()
	fmt.Println("Commands:")
	fmt.Println("  install  - Install LOIC and dependencies")
	fmt.Println("  update   - Update LOIC to latest version")
	fmt.Println("  run      - Run LOIC application")
	fmt.Println()
	fmt.Println("Note: This script is optimized for Linux and Termux")
}

func runCommand(cmd string, args ...string) (string, error) {
	var stdout, stderr bytes.Buffer
	
	command := exec.Command(cmd, args...)
	command.Stdout = &stdout
	command.Stderr = &stderr
	
	err := command.Run()
	if err != nil {
		return stdout.String(), fmt.Errorf("error running command: %v, stderr: %s", err, stderr.String())
	}
	
	return stdout.String(), nil
}

func runCommandShell(cmd string) (string, error) {
	var stdout, stderr bytes.Buffer
	
	command := exec.Command("bash", "-c", cmd)
	command.Stdout = &stdout
	command.Stderr = &stderr
	
	err := command.Run()
	if err != nil {
		return stdout.String(), fmt.Errorf("error running command: %v, stderr: %s", err, stderr.String())
	}
	
	return stdout.String(), nil
}

func which(command string) bool {
	_, err := exec.LookPath(command)
	return err == nil
}

func isTermux() bool {
	prefix := os.Getenv("PREFIX")
	return strings.Contains(prefix, "com.termux")
}

func checkDependencies() {
	fmt.Println("Checking dependencies...")
	
	if isTermux() {
		// Termux dependencies
		fmt.Println("Detected Termux environment")
		
		// Update packages
		fmt.Println("Updating package list...")
		runCommand("pkg", "update")
		
		// Install required packages
		packages := []string{"git", "mono"}
		for _, pkg := range packages {
			if !which(pkg) {
				fmt.Printf("Installing %s...\n", pkg)
				runCommand("pkg", "install", "-y", pkg)
			}
		}
	} else {
		// Linux dependencies
		fmt.Println("Detected Linux environment")
		
		// Detect distribution
		distro := detectDistro()
		fmt.Printf("Distribution: %s\n", distro)
		
		var packages []string
		var installCmd string
		
		switch distro {
		case "ubuntu", "debian":
			installCmd = "apt-get"
			packages = []string{
				"git",
				"mono-devel",
				"mono-runtime",
				"libmono-system-windows-forms4.0-cil",
				"monodevelop",
			}
		case "fedora":
			installCmd = "dnf"
			packages = []string{
				"git",
				"mono-devel",
				"mono-tools",
				"mono-basic",
			}
		default:
			fmt.Println("Unknown distribution. Installing base packages...")
			if which("apt-get") {
				installCmd = "apt-get"
				packages = []string{"git", "mono-devel", "mono-runtime"}
			} else if which("dnf") {
				installCmd = "dnf"
				packages = []string{"git", "mono-devel", "mono-runtime"}
			} else {
				fmt.Println("Please install manually: git, mono")
				return
			}
		}
		
		// Install packages with sudo
		for _, pkg := range packages {
			if !which(pkg) && pkg != "monodevelop" && pkg != "mono-tools" {
				fmt.Printf("Installing %s...\n", pkg)
				runCommand("sudo", installCmd, "install", "-y", pkg)
			}
		}
	}
	
	fmt.Println("✅ Dependencies check complete")
}

func detectDistro() string {
	// Try to detect distribution
	releaseFiles := []string{
		"/etc/os-release",
		"/etc/lsb-release",
		"/etc/debian_version",
		"/etc/fedora-release",
	}
	
	for _, file := range releaseFiles {
		if _, err := os.Stat(file); err == nil {
			content, err := os.ReadFile(file)
			if err == nil {
				text := strings.ToLower(string(content))
				if strings.Contains(text, "ubuntu") {
					return "ubuntu"
				}
				if strings.Contains(text, "debian") {
					return "debian"
				}
				if strings.Contains(text, "fedora") {
					return "fedora"
				}
			}
		}
	}
	
	// Default to ubuntu
	return "ubuntu"
}

func isLOIC() bool {
	if _, err := os.Stat(filepath.Join(loicDir, ".git")); err == nil {
		return true
	}
	if _, err := os.Stat(filepath.Join(loicDir, "LOIC", ".git")); err == nil {
		return true
	}
	return false
}

func getLOIC() error {
	fmt.Println("Cloning LOIC repository...")
	
	if isLOIC() {
		fmt.Println("LOIC already exists.")
		return nil
	}
	
	// Try git clone
	cmd := fmt.Sprintf("git clone %s -b %s", gitRepo, gitBranch)
	if isTermux() {
		cmd = fmt.Sprintf("git clone %s -b %s", gitRepo, gitBranch)
	}
	
	_, err := runCommandShell(cmd)
	if err != nil {
		return fmt.Errorf("failed to clone repository: %v", err)
	}
	
	return nil
}

func findSrcDir() string {
	possiblePaths := []string{
		filepath.Join(loicDir, "src"),
		filepath.Join(loicDir, "LOIC", "src"),
		filepath.Join(loicDir, "LOIC", "LOIC", "src"),
	}
	
	for _, path := range possiblePaths {
		if _, err := os.Stat(filepath.Join(path, "LOIC.sln")); err == nil {
			return path
		}
	}
	
	return ""
}

func findExePath(srcDir string) string {
	if srcDir == "" {
		return ""
	}
	
	possibleExe := []string{
		filepath.Join(srcDir, "bin", "Debug", "LOIC.exe"),
		filepath.Join(srcDir, "bin", "Release", "LOIC.exe"),
		filepath.Join(srcDir, "LOIC.exe"),
	}
	
	for _, exe := range possibleExe {
		if _, err := os.Stat(exe); err == nil {
			return exe
		}
	}
	
	return possibleExe[0]
}

func compileLOIC() error {
	fmt.Println("Compiling LOIC...")
	
	if err := getLOIC(); err != nil {
		return err
	}
	
	srcDir := findSrcDir()
	if srcDir == "" {
		return fmt.Errorf("could not find LOIC source directory")
	}
	
	fmt.Printf("Source directory: %s\n", srcDir)
	
	// Change to src directory
	originalDir, _ := os.Getwd()
	defer os.Chdir(originalDir)
	
	if err := os.Chdir(srcDir); err != nil {
		return fmt.Errorf("failed to change directory: %v", err)
	}
	
	// Try different compilation methods
	compileMethods := []struct {
		name    string
		command string
	}{
		{"MSBuild", "msbuild /p:TargetFrameworkVersion=v4.0 /p:Configuration=Debug"},
		{"XBuild", "xbuild /p:TargetFrameworkVersion=v4.0"},
		{"MCS/Direct", "mcs -target:winexe -out:LOIC.exe *.cs -r:System.Windows.Forms.dll -r:System.dll -r:System.Drawing.dll"},
	}
	
	for _, method := range compileMethods {
		fmt.Printf("Trying: %s\n", method.name)
		
		// Split command
		parts := strings.Fields(method.command)
		if len(parts) == 0 {
			continue
		}
		
		cmd := parts[0]
		args := parts[1:]
		
		if which(cmd) {
			_, err := runCommand(cmd, args...)
			if err == nil {
				// Check if executable was created
				exePath := findExePath(srcDir)
				if _, err := os.Stat(exePath); err == nil {
					fmt.Printf("✅ Compilation successful! EXE: %s\n", exePath)
					return nil
				}
			}
		}
	}
	
	// Try direct compilation with mcs on all .cs files
	fmt.Println("Trying full compilation with mcs...")
	
	// Find all .cs files
	pattern := "*.cs"
	matches, err := filepath.Glob(filepath.Join(srcDir, "**", pattern))
	if err == nil && len(matches) > 0 {
		var csFiles []string
		for _, match := range matches {
			if !strings.Contains(match, "bin") && !strings.Contains(match, "obj") {
				csFiles = append(csFiles, match)
			}
		}
		
		if len(csFiles) > 0 {
			args := []string{
				"-target:winexe",
				"-out:LOIC.exe",
				"-r:System.Windows.Forms.dll",
				"-r:System.dll",
				"-r:System.Drawing.dll",
				"-r:System.Net.dll",
				"-r:System.Xml.dll",
			}
			args = append(args, csFiles...)
			
			_, err := runCommand("mcs", args...)
			if err == nil {
				exePath := filepath.Join(srcDir, "LOIC.exe")
				if _, err := os.Stat(exePath); err == nil {
					fmt.Printf("✅ Compilation successful! EXE: %s\n", exePath)
					return nil
				}
			}
		}
	}
	
	return fmt.Errorf("all compilation methods failed")
}

func downloadLOICZip() error {
	fmt.Println("Downloading pre-compiled LOIC...")
	
	// Try to get latest release
	url := "https://github.com/NewEraCracker/LOIC/releases/latest/download/LOIC.zip"
	zipPath := filepath.Join(homeDir, "LOIC.zip")
	
	// Download
	resp, err := http.Get(url)
	if err != nil {
		return fmt.Errorf("download failed: %v", err)
	}
	defer resp.Body.Close()
	
	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("download failed with status: %s", resp.Status)
	}
	
	// Create file
	out, err := os.Create(zipPath)
	if err != nil {
		return err
	}
	defer out.Close()
	
	// Copy data
	_, err = io.Copy(out, resp.Body)
	if err != nil {
		return err
	}
	
	fmt.Printf("Downloaded to: %s\n", zipPath)
	
	// Extract
	fmt.Println("Extracting...")
	os.MkdirAll(loicDir, 0755)
	
	reader, err := zip.OpenReader(zipPath)
	if err != nil {
		return err
	}
	defer reader.Close()
	
	for _, file := range reader.File {
		path := filepath.Join(loicDir, file.Name)
		
		if file.FileInfo().IsDir() {
			os.MkdirAll(path, 0755)
			continue
		}
		
		dir := filepath.Dir(path)
		os.MkdirAll(dir, 0755)
		
		src, err := file.Open()
		if err != nil {
			continue
		}
		defer src.Close()
		
		dst, err := os.Create(path)
		if err != nil {
			continue
		}
		defer dst.Close()
		
		io.Copy(dst, src)
	}
	
	// Clean up
	os.Remove(zipPath)
	fmt.Println("✅ Extraction complete")
	
	return nil
}

func installLOIC() {
	fmt.Println("📦 Installing LOIC...")
	
	checkDependencies()
	
	// Try different installation methods
	methods := []struct {
		name string
		fn   func() error
	}{
		{"Download pre-compiled", downloadLOICZip},
		{"Build from source", compileLOIC},
	}
	
	for _, method := range methods {
		fmt.Printf("\n🔧 Trying: %s\n", method.name)
		if err := method.fn(); err == nil {
			// Check if we have an executable
			exe := findExePath(findSrcDir())
			if _, err := os.Stat(exe); err == nil {
				fmt.Printf("\n✅ Installation successful!\n")
				fmt.Printf("📁 LOIC installed at: %s\n", loicDir)
				fmt.Printf("📄 Executable: %s\n", exe)
				fmt.Printf("\nRun with: go run loic.go run\n")
				return
			} else {
				fmt.Println("⚠️  Installation completed but no executable found")
			}
		} else {
			fmt.Printf("❌ Method failed: %v\n", err)
		}
	}
	
	fmt.Println("\n❌ All installation methods failed")
	fmt.Println("\nManual installation steps:")
	fmt.Println("1. Install mono: pkg install mono (Termux) or apt-get install mono-devel (Linux)")
	fmt.Println("2. Clone repository: git clone https://github.com/NewEraCracker/LOIC.git")
	fmt.Println("3. Try compiling: cd LOIC/src && mcs -target:winexe -out:LOIC.exe *.cs -r:System.Windows.Forms.dll")
}

func runLOIC() {
	fmt.Println("Running LOIC...")
	
	// Check if mono exists
	if !which("mono") {
		fmt.Println("❌ Mono not found. Please install mono first.")
		if isTermux() {
			fmt.Println("  pkg install mono")
		} else {
			fmt.Println("  sudo apt-get install mono-runtime (Ubuntu/Debian)")
			fmt.Println("  sudo dnf install mono-runtime (Fedora)")
		}
		return
	}
	
	// Find executable
	srcDir := findSrcDir()
	if srcDir == "" {
		fmt.Println("❌ LOIC source directory not found.")
		fmt.Println("Please run 'install' first: go run loic.go install")
		return
	}
	
	exePath := findExePath(srcDir)
	if _, err := os.Stat(exePath); err != nil {
		fmt.Println("❌ LOIC executable not found.")
		fmt.Println("Please run 'install' first: go run loic.go install")
		return
	}
	
	fmt.Printf("Executable: %s\n", exePath)
	
	// Change to executable directory
	exeDir := filepath.Dir(exePath)
	os.Chdir(exeDir)
	
	fmt.Println("\n" + strings.Repeat("=", 50))
	fmt.Println("LOIC is starting...")
	fmt.Println("Press Ctrl+C to stop")
	fmt.Println(strings.Repeat("=", 50) + "\n")
	
	// Try different mono runtime versions
	commands := [][]string{
		{"mono", "--runtime=v4.0.30319", filepath.Base(exePath)},
		{"mono", filepath.Base(exePath)},
	}
	
	for _, cmd := range commands {
		fmt.Printf("Running: %s\n", strings.Join(cmd, " "))
		execCmd := exec.Command(cmd[0], cmd[1:]...)
		execCmd.Stdout = os.Stdout
		execCmd.Stderr = os.Stderr
		
		if err := execCmd.Run(); err == nil {
			break
		}
	}
	
	// Return to original directory
	os.Chdir(homeDir)
}

func updateLOIC() {
	fmt.Println("Updating LOIC...")
	
	if isLOIC() {
		if _, err := os.Stat(filepath.Join(loicDir, ".git")); err == nil {
			fmt.Println("Pulling latest changes...")
			os.Chdir(loicDir)
			runCommand("git", "pull", "--rebase")
			os.Chdir(homeDir)
			
			fmt.Println("Recompiling...")
			compileLOIC()
			return
		}
	}
	
	fmt.Println("LOIC not found in git repository. Reinstalling...")
	os.RemoveAll(loicDir)
	installLOIC()
}

File: loic.py
#!/usr/bin/env python3
# LOIC Installer for Termux (Android)
# Adapted for Termux environment

import os
import sys
import subprocess
import shutil
from pathlib import Path

# Constants
GIT_REPO = "https://github.com/NewEraCracker/LOIC.git"
GIT_BRANCH = "master"

# Termux paths
HOME = Path(os.environ.get("HOME", "/data/data/com.termux/files/home"))
LOIC_BASE = HOME / "LOIC"

def find_loic_src():
    """Find the actual LOIC src directory"""
    possible_paths = [
        LOIC_BASE / "src",
        LOIC_BASE / "LOIC" / "src",
        LOIC_BASE / "LOIC" / "LOIC" / "src",
    ]
    
    for path in possible_paths:
        if path.exists() and (path / "LOIC.sln").exists():
            return path
    return None

def find_exe_path(src_dir):
    """Find the LOIC executable path"""
    if src_dir is None:
        return None
    
    possible_exe = [
        src_dir / "bin" / "Debug" / "LOIC.exe",
        src_dir / "bin" / "Release" / "LOIC.exe",
        src_dir / "LOIC" / "bin" / "Debug" / "LOIC.exe",
        src_dir / "LOIC" / "bin" / "Release" / "LOIC.exe",
    ]
    
    for exe in possible_exe:
        if exe.exists():
            return exe
    return possible_exe[0]  # Return default path

def run_command(cmd, check=False, capture=False):
    """Run a shell command"""
    try:
        if capture:
            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
            return result.stdout.strip() if result.returncode == 0 else None
        else:
            subprocess.run(cmd, shell=True, check=False)
            return True
    except Exception as e:
        print(f"Error: {e}")
        return False

def which(command):
    """Check if command exists"""
    return shutil.which(command) is not None

def detect_termux():
    """Check if running in Termux"""
    return "com.termux" in os.environ.get("PREFIX", "")

def install_package(pkg):
    """Install package in Termux"""
    print(f"Installing {pkg}...")
    run_command(f"pkg install -y {pkg}")

def check_dependencies():
    """Check and install dependencies for Termux"""
    print("Checking dependencies for Termux...")
    
    # Update package list
    run_command("pkg update")
    
    # Install required packages
    packages = ["git", "mono", "msbuild", "mono-msbuild"]
    
    for pkg in packages:
        if not which(pkg):
            install_package(pkg)
    
    # Check if we have build tools
    if not which("msbuild") and not which("xbuild"):
        print("Installing mono-msbuild...")
        install_package("mono-msbuild")
        install_package("mono")
    
    print("Dependencies check complete.")

def is_loic():
    """Check if LOIC repository exists"""
    # Check both possible locations
    if (LOIC_BASE / ".git").exists():
        return True
    if (LOIC_BASE / "LOIC" / ".git").exists():
        return True
    return False

def get_loic():
    """Clone LOIC repository"""
    if not which("git"):
        install_package("git")
    
    if is_loic():
        print("LOIC already exists.")
        return
    
    print(f"Cloning LOIC from {GIT_REPO}...")
    os.chdir(HOME)
    run_command(f"git clone {GIT_REPO} -b {GIT_BRANCH}")
    
    # Check if cloned successfully
    if not is_loic():
        print("Error: Failed to clone LOIC repository.")
        return False
    
    # Find and show the structure
    src_dir = find_loic_src()
    if src_dir:
        print(f"Found LOIC source at: {src_dir}")
    else:
        print("Warning: LOIC source directory not found.")
    
    return True

def compile_loic():
    """Compile LOIC for Termux"""
    # Get LOIC first
    get_loic()
    
    # Find src directory
    src_dir = find_loic_src()
    if not src_dir:
        print("Error: Could not find LOIC source directory.")
        print("Looking in:", LOIC_BASE)
        print("Contents:", list(LOIC_BASE.glob("**/*.sln")))
        return False
    
    print(f"Compiling LOIC in: {src_dir}")
    os.chdir(src_dir.parent if src_dir.parent != src_dir else src_dir)
    
    # Try different build tools
    build_tools = []
    if which("msbuild"):
        build_tools.append("msbuild")
    if which("xbuild"):
        build_tools.append("xbuild")
    
    if not build_tools:
        print("Installing build tools...")
        install_package("mono-msbuild")
        if which("msbuild"):
            build_tools.append("msbuild")
        else:
            print("Error: No build tool available.")
            return False
    
    success = False
    for tool in build_tools:
        print(f"Trying {tool}...")
        
        # Try different build commands
        commands = [
            f"{tool} /p:TargetFrameworkVersion=v4.0 /p:Configuration=Debug",
            f"{tool} /p:Configuration=Debug",
            f"{tool} /p:TargetFrameworkVersion=v4.0",
            f"{tool}",
        ]
        
        for cmd in commands:
            print(f"  Running: {cmd}")
            run_command(cmd)
            
            # Check if exe was created
            exe_path = find_exe_path(src_dir)
            if exe_path and exe_path.exists():
                print(f"✅ Compilation successful! EXE found at: {exe_path}")
                success = True
                break
        
        if success:
            break
    
    if not success:
        print("❌ Compilation failed. Trying fallback method...")
        success = compile_loic_fallback(src_dir)
    
    return success

def compile_loic_fallback(src_dir):
    """Fallback compilation using mcs/dmcs directly"""
    print("Using fallback compilation method...")
    
    # Find all .cs files
    cs_files = list(src_dir.glob("**/*.cs"))
    if not cs_files:
        print("No .cs files found.")
        return False
    
    # Check for compiler
    compiler = None
    for cmd in ["dmcs", "mcs", "csc"]:
        if which(cmd):
            compiler = cmd
            break
    
    if not compiler:
        print("No C# compiler found.")
        return False
    
    # Compile all files
    output_path = src_dir / "LOIC.exe"
    refs = [
        "System.dll",
        "System.Windows.Forms.dll", 
        "System.Drawing.dll",
        "System.Net.dll",
        "System.Xml.dll"
    ]
    
    ref_args = " ".join([f"-r:{ref}" for ref in refs])
    cs_files_str = " ".join([str(f) for f in cs_files])
    
    cmd = f"{compiler} -target:winexe -out:{output_path} {ref_args} {cs_files_str}"
    print(f"Running: {cmd}")
    
    result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
    
    if output_path.exists():
        print(f"✅ Fallback compilation successful! EXE at: {output_path}")
        return True
    else:
        print("❌ Fallback compilation failed.")
        print("Error:", result.stderr[:500])
        return False

def run_loic():
    """Run LOIC application"""
    if not which("mono"):
        print("Mono not found. Installing...")
        install_package("mono")
    
    # Find source and exe
    src_dir = find_loic_src()
    if not src_dir:
        print("LOIC not found. Installing...")
        get_loic()
        src_dir = find_loic_src()
        if not src_dir:
            print("Error: Could not find LOIC.")
            return
    
    exe_path = find_exe_path(src_dir)
    if not exe_path or not exe_path.exists():
        print("LOIC executable not found. Compiling...")
        if not compile_loic():
            print("Error: Compilation failed.")
            return
        exe_path = find_exe_path(src_dir)
    
    if not exe_path or not exe_path.exists():
        print("Error: Could not find LOIC executable.")
        return
    
    print(f"Running LOIC from: {exe_path}")
    os.chdir(exe_path.parent)
    
    # Copy config if needed
    config_src = src_dir / "app.config"
    config_dst = exe_path.parent / "LOIC.exe.config"
    if config_src.exists() and not config_dst.exists():
        shutil.copy2(config_src, config_dst)
        print(f"Copied config to {config_dst}")
    
    print("\n" + "="*50)
    print("Starting LOIC...")
    print("Press Ctrl+C to stop")
    print("="*50 + "\n")
    
    try:
        # Try with different mono runtime versions
        commands = [
            f"mono --runtime=v4.0.30319 {exe_path.name}",
            f"mono {exe_path.name}"
        ]
        
        for cmd in commands:
            print(f"Running: {cmd}")
            result = subprocess.run(cmd, shell=True)
            if result.returncode == 0:
                break
            
    except KeyboardInterrupt:
        print("\n\nLOIC stopped.")
    except Exception as e:
        print(f"Error running LOIC: {e}")
    finally:
        os.chdir(HOME)

def update_loic():
    """Update LOIC from repository"""
    # Find the git repo
    git_dir = None
    if (LOIC_BASE / ".git").exists():
        git_dir = LOIC_BASE
    elif (LOIC_BASE / "LOIC" / ".git").exists():
        git_dir = LOIC_BASE / "LOIC"
    
    if git_dir:
        print(f"Updating LOIC at {git_dir}...")
        os.chdir(git_dir)
        run_command("git pull --rebase")
        os.chdir(HOME)
        compile_loic()
    else:
        print("LOIC repository not found. Cloning...")
        get_loic()
        compile_loic()

def print_usage():
    """Print usage information"""
    print("LOIC Installer for Termux")
    print("=" * 40)
    print()
    print("Usage: python3 loic.py <install|update|run>")
    print()
    print("Commands:")
    print("  install  - Install LOIC and dependencies")
    print("  update   - Update LOIC to latest version")
    print("  run      - Run LOIC application")
    print()
    print("Note: This script is optimized for Termux (Android)")

def main():
    """Main function"""
    if len(sys.argv) != 2:
        print_usage()
        sys.exit(1)
    
    if not detect_termux():
        print("Warning: Not running in Termux environment.")
        print("This script is optimized for Termux on Android.")
        response = input("Continue anyway? (y/N): ")
        if response.lower() != 'y':
            sys.exit(0)
    
    command = sys.argv[1].lower()
    
    if command == "install":
        print("📦 Installing LOIC for Termux...")
        check_dependencies()
        if compile_loic():
            print("\n✅ Installation successful!")
            print("Run with: python3 loic.py run")
        else:
            print("\n❌ Installation failed.")
            print("Please check errors above.")
            print("You may need to install mono manually:")
            print("  pkg install mono mono-msbuild")
            
    elif command == "update":
        update_loic()
        
    elif command == "run":
        run_loic()
        
    else:
        print(f"Unknown command: {command}")
        print_usage()
        sys.exit(1)

if __name__ == "__main__":
    main()
